packages/graphql/lib/src/core/query_options.dart

```diff
@@ -7,6 +9,9 @@ import 'package:gql/ast.dart';
 import 'package:graphql/client.dart';
 import 'package:meta/meta.dart';
 
+typedef OnQueryCompleted = FutureOr<void> Function(dynamic data);
```

Comment:
Also, as I found out in #1099, this should be

```suggestion
typedef OnQueryCompleted = FutureOr<void> Function(Map<String, dynamic> data);
```